tests/src/Functional/NewsPageTest.php

```diff
@@ -77,24 +84,95 @@ public function testNewsFields() {
   }
 
   /**
-   * Pathauto and breadcrumbs.
+   * News article, newsroom, featured news.
    */
-  public function testNewsPaths() {
-    $this->createNode([
+  public function testNewsPages() {
+    $body = $this->randomMachineName(64);
+    $newsArticle = $this->createNode([
       'title' => 'News article 1',
+      'body' => $body,
       'type' => 'localgov_news_article',
       'status' => NodeInterface::PUBLISHED,
     ]);
 
-    $this->drupalGet('news');
+    $this->drupalGet('news/' . date('Y') . '/news-article-1');
     $this->assertText('News article 1');
+    $this->assertText($body);
+
+    $newsroom = $this->createNode([
+      'title' => 'News',
+      'type' => 'localgov_newsroom',
+      'path' => [
+        'alias' => '/news',
+      ],
+      'status' => NodeInterface::PUBLISHED,
+    ]);
 
-    $this->drupalGet('news/' . date('Y') . '/news-article-1');
+    $this->drupalGet('news');
     $this->assertText('News article 1');
 
-    $trail = ['' => 'Home'];
-    $trail += ['news' => 'News'];
-    $this->assertBreadcrumb(NULL, $trail);
+    // Add News article 1 to the featured news block.
+    $newsroom->set('localgov_newsroom_featured', $newsArticle->id());
+    $newsroom->save();
+    drupal_flush_all_caches();
+    $this->drupalGet('news');
+
+    // Test the Featured news block displays.
+    $this->assertSession()->elementExists('css', 'div#block-localgov-featured-news-articles');
+
+    // Test that News article 1 is no longer included in the news listing.
+    $this->assertSession()->elementNotExists('css', 'div#infinite-scroll--wrapper article');
+
+  }
+
+  /**
+   * News default media.
+   */
+  public function testNewsMedia() {
+    // Image file for testing.
+    $imageData = file_get_contents('https://upload.wikimedia.org/wikipedia/en/a/a9/Example.jpg');
+    $destination = 'public://example.jpg';
+    $imageFile = file_save_data($imageData, $destination, FileSystemInterface::EXISTS_REPLACE);
+    $this->assertTrue(file_exists($destination));
+
+    // Create default news image entity using example file.
+    $newsMedia = Media::create([
+      'name' => 'Example',
+      'bundle' => 'localgov_news_default_image',
+      'field_media_image' => [
+        'target_id' => $imageFile->id(),
+        'alt' => 'Alternative text',
+      ],
+    ]);
+    $newsMedia->setPublished()->save();
```

Comment:
Could we also test the select from a list functionalty (if thats a new issue I'll raise it).

Reply:
That creates the media type item, it's selected from the list a bit further on:

https://github.com/localgovdrupal/localgov_news/blob/a4d6f93cb7ffe46fbb6d14ca26a29ac7414f8fc6/tests/src/Functional/NewsPageTest.php#L155-L160

Or am I misunderstanding?

Reply:
Sorry misunderstanding, can resolve this now.